src/numerics/petsc_matrix.C

```diff
@@ -1197,6 +1268,10 @@ void PetscMatrix<T>::matrix_matrix_mult (SparseMatrix<T> & X_in, SparseMatrix<T>
 
   ierr = MatMatMult(_mat, X->_mat, MAT_INITIAL_MATRIX, PETSC_DEFAULT, &Y->_mat);
   LIBMESH_CHKERR(ierr);
+
+  // make sure Y is initialized and close it
+  Y->_is_initialized = true;
+  Y->close();
```

Comment:
The trouble with throwing close() calls everywhere is that each one of them is a parallel communication, right?

This seems like a good place to do it (I'm not even sure what you can follow up a MatMatMult with if you don't close it in between), but at some point we really ought to get more rigorous about documenting which of our methods follow up with a close() and which don't.

Reply:
Thanks @roystgnr . It seems like removing the `Y->close()` is better here, according to @fdkong (https://github.com/libMesh/libmesh/pull/2836#discussion_r560400499).  I tried it and found no issue. (I accidentally made duplicated changes here and in #2836  and I have removed the change in the other PR. I will update this one soon)

I agree that we should have better documentation about when close() is necessary or not.